eduiddashboard/views/mobiles.py

```diff
@@ -116,16 +116,14 @@ def remove_action(self, index, post_data):
     def setprimary_action(self, index, post_data):
         mobiles = self.user.get_mobiles()
 
-        if index > len(mobiles):
-            message = _("That mobile phone number doesn't exists")
-            return {
-                'result': 'bad',
-                'message': get_localizer(self.request).translate(message),
-            }
+        try:
+            mobile = self.user.get_mobiles()[index]
+        except IndexError:
+            return self.sync_user()
 
-        if index > len(mobiles):
-            message = _("You need to verify that mobile phone number "
-                        "before be able to set as primary")
+        if not mobile.get('verified', False):
+            message = _('You need to confirm your mobile number '
+                        'before it can become primary')
```

Comment:
maybe we should revert the message text change here, or we will need to sync up the translation accordingly.